tools/compile_tl_map.php

```diff
@@ -21,9 +21,38 @@ function getMapNodes(string $filename, array &$map)
     scandir($mapStore),
     ['.', '..']
 );
-$resultMap = [];
+
+// find layer_compiled_XXX.json with max layer and sort simple layer files (layer_XXX.json) by version
+$compiledLayerFiles = [];
+$simpleLayerFiles = [];
 foreach ($maps as $mapFile) {
-    $map = getMapNodes($mapStore.'/'.$mapFile, $resultMap);
+    preg_match("/^layer_compiled_(\d+)\.json$/i", $mapFile, $matches);
+    if (sizeof($matches) > 0) {
+        $compiledLayerFiles[$matches[1]] = $mapFile;
+    }
+    preg_match("/^layer_(\d+)\.json$/i", $mapFile, $matches);
+    if (sizeof($matches) > 0) {
+        $simpleLayerFiles[$matches[1]] = $mapFile;
+    }
+}
+krsort($compiledLayerFiles);
+ksort($simpleLayerFiles);
+
+$resultMap = ['constructors' => [], 'methods' => []];
+$lastLayer = 0;
+// last compiled layer file is the most recent
+foreach ($compiledLayerFiles as $layer => $mapFile) {
```

Comment:
Is there any precedents when same TL ID appeared in new Layer with new structure? Why prioritize then?

Reply:
TL object can change only field, no id. We must compile by version number to get right layer